packages/arb-bridge-eth/contracts/validator/GasRefunder.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2021, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity ^0.8.7;
+
+import "./IGasRefunder.sol";
+
+import "@openzeppelin/contracts-0.8/access/Ownable.sol";
+
+contract GasRefunder is IGasRefunder, Ownable {
+    mapping(address => bool) public allowedContracts;
+    mapping(address => uint256) public lastContractRefund;
+    uint256 public maxRefundeeBalance;
+    uint256 public extraGasMargin;
+
+    event RefundedGasCosts(
+        address indexed refundee,
+        address indexed contractAddress,
+        uint256 gas,
+        uint256 amountPaid,
+        bool success
```

Comment:
Since we have a spare param to index, should we add `success`?
Makes it easy for us to check if there were any failed refunds